path.cabal

```diff
@@ -74,21 +74,15 @@ library
                    , deepseq
                    , exceptions >= 0.4     && < 0.11
                    , hashable   >= 1.2     && < 1.5
-                   , path:compat
+                   , path:os-string-compat
                    , text
                    , template-haskell
+                   , filepath   >= 1.4.100.0
 
-  if flag(old-os-string)
-    build-depends:   filepath   >= 1.4.100.0
-                   , os-string  < 2.0.0
-  else
-    build-depends:   filepath   >= 1.5.0.0
-                   , os-string  >= 2.0.0
-
-library compat
+library os-string-compat
```

Comment:
Are you sure this will be stack-compatible? That's a must.

Reply:
Yes, it is; Stack supports depending on (public) sublibraries (of other packages) since 2.15.1; Not sure since which version depending on private sub-libraries is supported...